### DISCUSSION TITLE
替换版面分析模型

### DISCUSSION THREAD

**Discussion by Andy197527** [PaddlePaddle/PaddleOCR#13351] [Q&A]
您好，在这里我想咨询一些问题，就是关于PPstructureV2里边的版面分析模型的问题，我自己根据你们的方法训练好了一个版面分析模型，但是我却无法替换上去，就是替换掉原先的版面分析模型。我自己训练的版面分析模型也能推理，但是一替换就出错，也咨询了好多人，但是仍然无法解决，特意来咨询一下。
用的picodet_lcnet_x1_0_fgd_layout_cdla训练的，希望能给予解答。一直出现的错误是：ValueError: (InvalidArgument) The input of Op(Conv) should be a 4-D or 5-D Tensor. But received: input's dimension is -1, input's shape is [].
  [Hint: Expected in_dims.size() == 4 || in_dims.size() == 5 == true, but received in_dims.size() == 4 || in_dims.size() == 5:0 != true:1.] (at ..\paddle/phi/kernels/cpu/conv_util.h:137)
  [operator < fused_conv2d_add_act > error]

**Comment by GreatV** [comment]
是不是没有固定shape呀

**Reply by Andy197527** [reply]
> 是不是没有固定shape呀

能麻烦的详细说一说嘛？如果知道原因麻烦请详细的说一说，我最近因为这个头疼死了，找不到原因。我训练出来的参数是
mode: paddle
draw_threshold: 0.5
metric: VOC
use_dynamic_shape: false
arch: GFL
min_subgraph_size: 3
Preprocess:
- interp: 2
  keep_ratio: false
  target_size:
  - 800
  - 608
  type: Resize
- is_scale: true
  mean:
  - 0.485
  - 0.456
  - 0.406
  std:
  - 0.229
  - 0.224
  - 0.225
  type: NormalizeImage
- type: Permute
- stride: 32
  type: PadStride
label_list:
- Text
- Table
- Figure
NMS:
  keep_top_k: 100
  name: MultiClassNMS
  nms_threshold: 0.5
  nms_top_k: 1000
  score_threshold: 0.3
fpn_stride:
- 8
- 16
- 32
- 64
其中arch: GFL是跟官网上的不一样的，官网的是arch: PicoDet。

**Reply by Andy197527** [reply]
> 是不是没有固定shape呀

这个固定shape在哪设置呢？

**Reply by lanheshui** [reply]
请问你解决了吗

**Comment by GreatV** [comment]
你描述的太抽象了，什么环境，怎么替换的，有没有可复现的demo都没有

**Comment by crackso** [comment]
你用的动态图模型还是静态图模型？

**Reply by lanheshui** [reply]
静态图，我把导出模型架构GFL换成PicoDet解决了
